### PR TITLE
Refactor user-defined meta to be tracked on the service instead of task

### PR DESCRIPTION
**Pull request by findkim**
This omits the need for the `UserDefinedMeta()` getter and reduces an attribute from the `driver.Task` object

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -271,16 +271,17 @@ func newDriverTasks(conf *config.Config, providerConfigs driver.TerraformProvide
 // configuration by ID to a driver service type. If a service is not
 // explicitly configured, it assumes the service is a logical service name
 // in the default namespace.
-func getService(services *config.ServiceConfigs, id string) driver.Service {
+func getService(services *config.ServiceConfigs, id string, meta map[string]map[string]string) driver.Service {
```
Comment (eikenb) [inline controller/controller.go:274]: Those map of maps data types can really get verbose. I tend to want to make them into local types to help with readability. Definitely not a big deal though.

**Review by eikenb — APPROVED** [review]
Nice simplification.

File: driver/task.go [hunk 2]
```diff
@@ -226,21 +224,6 @@ func (t *Task) Source() string {
...
-	t.mu.RLock()
-	defer t.mu.RUnlock()
-
-	meta := make(map[string]map[string]string)
-	for k, nestedMap := range t.userDefinedMeta {
-		meta[k] = make(map[string]string)
-		for nestedK, nestedV := range nestedMap {
-			meta[k][nestedK] = nestedV
-		}
-	}
```
Comment (lornasong) [inline driver/task.go:240]: Now that user-defined-meta is a field on services, does this type of 'copying' logic have to be added to the Services() getter, or maybe it's ok as is?

**Review by lornasong — APPROVED** [review]
looks good to me as well! left a non-blocking question

File: driver/task.go (hunk 2 above)
Comment (findkim) [reply driver/task.go:240]: Believe so since maps are by reference, good catch.

Comment (findkim) [reply driver/task.go:240]: Love me some go playground https://play.golang.org/p/L-KljbPqTqZ

File: controller/controller.go (hunk 1 above)
Comment (findkim) [reply controller/controller.go]: :nod: i do not like either. Thanks for the nudge, I looked around and saw this nested map in a number of places. Time to typify this! https://github.com/hashicorp/consul-terraform-sync/pull/304/commits/1e686da0eb8c13fa8a9b67d03839d16a0db6f3de

**Pull request merged**